clients/tfchain-client-go/go.mod

```diff
@@ -1,16 +1,16 @@
 module github.com/threefoldtech/tfchain/clients/tfchain-client-go
 
-go 1.17
+go 1.21
```

Comment:
Is there a particular reason to update the MSV?
Given that there haven't been any significant changes, I assume we don't this.